katsdpimageutils/test/test_primary_beam_correction.py

```diff
@@ -0,0 +1,111 @@
+"""Tests for :mod:`katsdpimageutils.primary_beam_correction`."""
+import shutil
+import os
+
+import numpy as np
+from astropy.io import fits
+from nose.tools import assert_equal
+
+from ..primary_beam_correction import _get_value_from_history, weighted_average, inverse_variance, standard_deviation, write_new_fits # noqa E501 
+
+
+class TestStandardDeviation():
+    def setup(self):
+        self.arr = np.array([2, 4, 6, 8, 10]).astype(float)
+
+    def test_nonans(self):
+        data = self.arr
+        med, sd = standard_deviation(data)
+        assert_equal([med, sd], [6.0, 2.9652])
+
+    def test_nans(self):
+        data = self.arr
+        data[3] = np.nan
+        med, sd = standard_deviation(data)
+        assert_equal([med, sd], [5.0, 2.9652])
+
+    def test_allnan(self):
+        data = self.arr
+        data[:] = np.nan
+        with np.testing.assert_warns(RuntimeWarning):
+            med, sd = standard_deviation(data)
+
+
+class TestInverseVariance():
+    def setup(self):
+        self.data = np.array([2, 4, 6, 8, 10]).astype(float)
+
+    def test_all_zero(self):
+        data = self.data
+        data[:] = 0.0
+        inv_var = inverse_variance(data)
+        assert_equal(inv_var, 0.0)
+
+    def test_all_nans(self):
+        data = self.data
+        data[:] = np.nan
+        with np.testing.assert_warns(RuntimeWarning):
+            inverse_variance(data)
+
+    def test_nonans(self):
+        data = self.data
+        inv_var = inverse_variance(data)
+        assert_equal(inv_var, 1/(2.9652)**2)
+
+    def test_cut_5sd(self):
```

Comment:
I see in the docstring for `inverse_variance` it says it should reject zeros as well. Perhaps you should add a test (or a zero in this test) to check that this is functioning as expected

Reply:
Done.